control_protocol.md

```diff
@@ -0,0 +1,296 @@
+    N1 ->> CB: N1.CB|N1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> N1: N1.CA|N1.CB| Property A has value 5.
+    N1 ->> CA: N1.CA|N1.CB| Property A has value 5.
+:::
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> N1: N2.CB|N1.CA| Give me property A.
+    participant N1 as N1.COORDINATOR
+    Note over N1,N2: N1 DEALER socket sends to N2 ROUTER
+    participant N2 as N2.COORDINATOR
+    N1 ->> N2: N2.CB|N1.CA| Give me property A.
+    N2 ->> CB: N2.CB|N1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> N2: N1.CA|N2.CB| Property A has value 5.
+    Note over N1,N2: N2 DEALER socket sends to N1 ROUTER
+    N2 ->> N1: N1.CA|N2.CB| Property A has value 5.
+    N1 ->> CA: N1.CA|N2.CB| Property A has value 5.
+:::
+
+Prerequisite of Communication between two Components are:
+- Both Components are connected to a Coordinator and {ref}`signed in<sign-in>`.
+- Both Components are either connected to the same Coordinator (example one), or their Coordinators are connected to each other (example two).
+
+
+The following flow chart shows the decision scheme and message modification in a Coordinator `Co1` of Node `N1`.
+Its full ID is `N1.Coordinator`.
+`n0`, `nR` are placeholders for sender and recipient Node ID.
+`recipient` is a placeholder for the recipient Component ID.
+`IA` is the connection identity of the Component `N1.CA` (if it is directly connected to `Co1`) or of its Coordinator `n0.COORDINATOR`.
+`IB` is the connection identities of `N1.Recipient`.
+Bold arrows indicate message flow, thin lines indicate decision flow.
+Thin, dotted lines indicate decision flow in case of errors.
+Placeholder values are written in lowercase, while actually known values are begin with an uppercase letter.
+
+:::{mermaid}
+flowchart TB
+    C1([N1.CA DEALER]) == "nr.recipient|n0.CA|Content<br>(==nr.recipient|N1.CA|Content)" ==> R0
+    C0([n0.COORDINATOR DEALER]) == "nr.recipient|n0.CA|Content" ==> R0
+    R0[receive] == "IA|nr.recipient|n0.CA|Content" ==> Cn0{n0 == N1?}
+    Cn0-->|no| RemIdent
+    Cn0-->|yes| Clocal{CA in <br>local address book?}
+    Clocal -->|yes| CidKnown{IA is CA's identity<br> in address book?}
+    CidKnown -->|yes| RemIdent
+    Clocal -.->|no| E1[ERROR: Sender did not sign in] ==>|"IA|n0.CA|N1.COORDINATOR|ERROR: Sender dit not sign in<br>(==IA|N1.CA|N1.COORDINATOR|ERROR...)"| S
+    S[send] ==> WA([N1.CA DEALER])
+    CidKnown -.->|no| E2[ERROR: ID and identity do not match]==>|"IA|n0.CA|N1.COORDINATOR|ERROR: ID and identity do not match<br>(==IA|N1.CA|N1.COORDINATOR|ERROR...)"| S
+    RemIdent[remove sender identity] == "nr.recipient|n0.CA|Content" ==> Cnr
+    Cnr -- "is None" --> Local
+    Cnr{nr?} -- "== N1"--> Local
+    Local{recipient<br>==<br>COORDINATOR?} -- "yes" --> Self[Message for Co1<br> itself]
+    Self == "nr.recipient|n0.CA|Content<br>(==N1.COORDINATOR|n0.CA|Content)" ==> SC([Co1 Message handling])
+    Local -- "no" --> Local2a{recipient in Address book?}
```

Comment:
I think the constellation "nr? is None" should trigger an error, as the namespace is missing -- we should _only_ allow that for the sign-in, and in that case the whole recipient is/could be/should be missing (discussed elsewhere).

Or am I confused about the decision to always use full "addresses"/paths, as all Components know their namespace after signin, anyway?